Added advice for future process group (#222)
aau-giraf/wiki: docs/process_manual/2020/advice.md

# Advice for Future Process Groups

This section will describe our views and experiences, and what we found out that worked. 
Previous advice can be found [here](../../getting_started/advice_for_future_giraf/process_advice.md).

## The Process in General

* The project is everyone's, the process and PO group does not have more authority than everyone else, just more responsibility.

> * Take your time to understand the process and project, read the wiki, it is well spent.

Reviewer: "it is well spent"? What is ment here? Do you mean: "It is worth the read." or so?

> * Use the process exactly as described on this wiki and then modify it, to avoid confusion.

Reviewer: "Use the process and then modify it"? Is it ment that they take the current process that exists, and then change it to what would fit their process?

* Use Gitflow, as described in the wiki.

> * Be aware of the process changeing and make an active decision about what to do.

Reviewer: ```suggestion
* Be aware of the process changing and make an active decision about what to do.
```

* Use the tools provided on this wiki and improve them for yourselves and the next years.
* Use some time at the end of your project making it ready to hand off to the next year.

>     * An important way to do this, is to keep Github clean of unused branches throughout the project.

Reviewer: ```suggestion
    * An important way to do this, is to keep GitHub clean of unused branches throughout the project.
```

## Communication

* Use Discord or Slack as the communication medium. Use it to broadcast event start times and deadlines.

> * Events should be held at a physical location with all developers present.

Reviewer: Physical and/or remote location maybe? Since we have heavily used Discord due to Covid-19.

    * Start the meeting with the agenda. Remind people what the purpose is. Do this **every** time.
    * Take short breaks between the different activities to keep the groups focused.
    * If you do activities in smaller groups eg. Retrospective, remember to keep the groups small (<8) if you can. 
    This prevents people from hiding behind others who speaks the entire time.
* Keep things transparent. Files should be public and so should the thoughts behind process changes.
* Respond to questions in timely manner. Do not be condescending or hostile in your answers. 
    * You need to repeat a lot of information because people will forget or be confused. 
* Always be clear on what you say. Do not expect people to understand you implicitly. 
* Discuss the responsibilites of the process and PO group with the other developers at the start of the project.

>     * Maybe try discussing the responsibilites before choosing who will act as them.

Reviewer: "before choosing who will act as them"? Is it ment: Who will carry out the responsibilities?

## Keeping an Overview

* Create a calendar available for all the groups in the project. Fill the calendar with all dates and times planned from the start.

> * Decide where all your documents should be stored and be sure to be transparent with the other developersteam with the documents.

Reviewer: ```suggestion
* Decide where all your documents should be stored and be sure to be transparent with the other developer teams with the documents.
```

> * Find an effective to monitor and handle pull requests and reviews. We used google sheets and Github's API to automate it. 

Reviewer: ```suggestion
* Find an effective to monitor and handle pull requests and reviews. We used Google sheets and GitHub's API to automate it. 
```

> * Clean up branches every sprint. Delete those that werent deleted when merging and make sure every branch has a purpose. 

Reviewer: Clean up branches is already written earlier, although I find it okay if it is there again to remind the reader.

> * Decide if you want META-groups. Groups focused on different aspects of the project. Front-end, backend, server and security.

Reviewer: ```suggestion
* Decide if you want META-groups, which are groups focused on different aspects of the project such as front-end, backend, server and security.
```

Reviewer 2: `front-end` and `backend` should also use the same spelling format. Either with or without dash.
